app/core/src/main/java/com/fsck/k9/Preferences.java

```diff
@@ -190,4 +194,43 @@ public Storage getStorage() {
     private BackendManager getBackendManager() {
         return DI.get(BackendManager.class);
     }
+
+    public void saveAccount(Account account) {
+        StorageEditor editor = storage.edit();
+
+        if (!accounts.containsKey(account.getUuid())) {
```

Comment:
Also new accounts are already added to `accounts` with their uuids here. This results in new accounts always having -1 as  accountNumber. 

Using `if (account.getAccountNumber() < 0) {` instead seems to work for me.

Reply:
Good catch! This previously checked for the uids in committed storage.

Hm. @cketti is there a reason why accountNumber is sometimes used instead of uuid, e.g. in [NewMailNotifications](https://github.com/k9mail/k-9/blob/c043eb0df4f79f450c46abaac089fcb68adb7992/app/core/src/main/java/com/fsck/k9/notification/NewMailNotifications.java#L107)?

Reply:
In that particular case the account's UUID probably would have done the trick. In other cases we need a number. `Account.getId()` might have been a better name, though.